src/performance/perfcollect/perfcollect

```diff
@@ -0,0 +1,560 @@
+#    started on already running processes.
+# 2. Run this script: sudo ./perfcollect collect samplePerfTrace
+#    This will start data collection.
+# 3. Let the repro run as long as you need to capture the performance problem.
+# 4. Hit CTRL+C to stop collection.
+#    When collection is stopped, the script will create a trace.tgz file matching the name specified on the
+#    command line.  This file will contain the trace, JIT-compiled symbol information, and all debugging
+#    symbols for binaries referenced by this trace that were available on the machine at collection time.
+#
+# How to view a performance trace:
+# 1. Run this script: ./perfcollect view samplePerfTrace.trace.tgz
+#    This will extract the trace, place and register all symbol files and JIT-compiled symbol information
+#    and start the perf_event viewer.  By default, you be looking at a callee view - stacks are ordered
+#    top down.  For a caller or bottom up view, specify '-graphtype caller'. 
+#############################################################################################################
+
+
+######################################
+## FOR DEBUGGING ONLY
+######################################
+# set -x
+
+######################################
+## Configuration
+######################################
+
+# The location of the perf executable.
+perfcmd=/usr/bin/perf
+
+######################################
+## Collection Options
+## NOTE: These values represent the collection defaults.
+######################################
+
+# Set when we parse command line arguments to determine if we should enable specific collection options.
+collect_cpu=1
+collect_threadTime=0
+
+# Declare an array of events to collect.
+declare -a eventsToCollect
+
+######################################
+## Helper Functions
+######################################
+
+##
+# Console text color modification helpers.
+##
+RedText()
+{
+	tput setaf 1
+}
+
+GreenText()
+{
+	tput setaf 2
+}
+
+ResetText()
+{
+	tput sgr0
+}
+
+# $1 == Status message
+WriteStatus()
+{
+	GreenText
+	echo $1
+	ResetText
+}
+
+# $1 == Message.
+FatalError()
+{
+	RedText
+	echo "ERROR: $1"
+	ResetText
+	PrintUsage
+	exit 1
+}
+
+######################################
+# Prerequisite Installation
+######################################
+IsUbuntu()
+{
+	local ubuntu=0
+	if [ -f /etc/lsb-release ]
+	then
+		local flavor=`cat /etc/lsb-release | grep DISTRIB_ID`
+		if [ "$flavor" == "DISTRIB_ID=Ubuntu" ]
+		then
+			ubuntu=1
+		fi
+	fi
+        
+	echo $ubuntu
+}
+
+InstallPerf_Ubuntu()
+{
+	# Disallow non-root users.
+	if [ `whoami` != "root" ]
+	then
+		RedText
+		echo "This script must be run as root."
+		ResetText
+		exit 1
+	fi
+
+	# Install packages.
+	GreenText
+	echo "Installing perf_event packages."
+	ResetText
+	apt-get install linux-tools-common linux-tools-`uname -r` linux-cloud-tools-`uname -r`
+}
+
+InstallPerf()
+{
+	if [ "$(IsUbuntu)" == "1" ]
```

Comment:
SupportsAutoInstall is all about determining whether or not we actually support it.  Currently, the script uses this to tell the user about the fact that they can use auto install.  However, once ai user attempts to auto install, we have to determine the distro that we're on so that we run the right commands.  For Ubuntu, you use apt-get install, but the various distro families use different package managers.